gateway-service-knoxssout/src/main/java/org/apache/knox/gateway/service/knoxsso/WebSSOutResource.java

```diff
@@ -106,6 +111,23 @@ private boolean removeAuthenticationToken(HttpServletResponse response) {
     }
     response.addCookie(c);
 
+    Cookie[] cookies = request.getCookies();
+    if (cookies != null) {
+      Optional<Cookie> ssoCookie = Arrays.stream(cookies).filter(cookie -> cookie.getName().equals(cookieName)).findFirst();
```

Comment:
Could you extract the cookie finding logic in its own method?

Something like this:

```java
Optional<Cookie> findCookie(String cookieName) {
    Cookie[] cookies = request.getCookies();
    if (cookies != null) {
      return Arrays.stream(cookies).filter(cookie -> cookie.getName().equals(cookieName)).findFirst();
   } else {
      return Optional.empty();
   }
}
```